app/lib/actions/katello/host/upload_profiles.rb

```diff
@@ -47,7 +47,15 @@ def run
               when "enabled_repos"
                 host.import_enabled_repositories(payload)
               else
-                host.import_module_streams(payload)
+                begin
+                  enabled_payload = payload.map do |profile|
+                    profile.slice("name","stream","version","context", "arch") if profile["status"] == "enabled"
+                  end.compact
+                  ::Katello::Pulp::Consumer.new(host.content_facet.uuid).upload_module_stream_profile(enabled_payload)
+                  host.import_module_streams(payload)
+                rescue RestClient::ResourceNotFound
+                  Rails.logger.warn("Host with ID %s was not known to Pulp, continuing" % host_id)
+                end
```

Comment:
this method is getting kinda ugly and big, do you mind extracting some of this into a method?